.github/workflows/codecov.yml

```
@@ -0,0 +1,22 @@
+name: Codecov
+on: [push, pull_request]
+
+jobs:
+  build:
+    name: Build
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: '^1.21'
+      - name: Run tests with coverage
+        run: go test ./... -coverprofile=coverage.txt -covermode=atomic
+      - name: Upload coverage to Codecov
+        uses: codecov/codecov-action@v2
```

```suggestion
        uses: codecov/codecov-action@v3
```

https://github.com/codecov/codecov-action